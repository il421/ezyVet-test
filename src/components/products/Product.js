import React from 'react';
import { formatCurrency } from '../../libs/currency.js';
import '../../styles/products/product.css';
import '../../styles/common/button.css';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';

const Product = ({ id, name, price, mediaUrl, addToCart }) => {
  return (
    <article id={ id } className="product">
      <img className="product__img" src={ mediaUrl } alt="Product description" />
      <div className="product__wrapper">
        <div className="product__name">{ name }</div>
        <div className="product__price">{ formatCurrency(price) }</div>
        <button
          className="product__add button"
          onClick={ () => addToCart({ id, name, price, qty: 1 }) }
        >
          <span>Add to cart</span>
          <FontAwesomeIcon icon="cart-plus" size="1x" />
        </button>
      </div>
    </article>
  );
};


export default Product;
