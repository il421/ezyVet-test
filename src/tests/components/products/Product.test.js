import React from 'react';
import { shallow } from 'enzyme';
import Product from '../../../components/products/Product.js';
import { products, cart } from '../../fixtures/data';

let wrapper, addToCart;

beforeEach(() => {
  addToCart = jest.fn();

  wrapper = shallow(
    <Product
      { ...products[0] }
      addToCart={ addToCart }
    />
  );
});

test('should render Product without crashing', () => {
  expect(wrapper).toMatchSnapshot();
});

test('should call addToCart if click Add To Cart', () => {
  wrapper.find('.product__add').simulate('click')
  expect(addToCart).toHaveBeenLastCalledWith(cart[0]);
});
