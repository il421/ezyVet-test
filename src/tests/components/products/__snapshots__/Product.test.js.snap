// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Product without crashing 1`] = `
<article
  className="product"
  id={1}
>
  <img
    alt="Product description"
    className="product__img"
    src="http://placehold.it/200x200"
  />
  <div
    className="product__wrapper"
  >
    <div
      className="product__name"
    >
      Sledgehammer
    </div>
    <div
      className="product__price"
    >
      $125.75
    </div>
    <button
      className="product__add button"
      onClick={[Function]}
    >
      <span>
        Add to cart
      </span>
      <FontAwesomeIcon
        border={false}
        className=""
        fixedWidth={false}
        flip={null}
        icon="cart-plus"
        inverse={false}
        listItem={false}
        mask={null}
        pull={null}
        pulse={false}
        rotation={null}
        size="1x"
        spin={false}
        symbol={false}
        title=""
        transform={null}
      />
    </button>
  </div>
</article>
`;
